ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala

```diff
@@ -80,21 +81,34 @@ trait WriteService
     * time for submitting and validating large transactions before they are
     * timestamped with their record time.
     *
-    * @param submitterInfo   : the information provided by the submitter for
-    *                        correlating this submission with its acceptance or rejection on the
-    *                        associated [[ReadService]].
-    * @param transactionMeta : the meta-data accessible to all consumers of the
-    *                        transaction. See [[TransactionMeta]] for more information.
-    * @param transaction     : the submitted transaction. This transaction can contain local
-    *                        contract-ids that need suffixing. The participant state may have to
-    *                        suffix those contract-ids in order to guaranteed their global
-    *                        uniqueness. See the Contract Id specification for more detail
-    *                        daml-lf/spec/contract-id.rst.
+    * @param submitterInfo               the information provided by the submitter for
+    *                                    correlating this submission with its acceptance or rejection on the
+    *                                    associated [[ReadService]].
+    * @param transactionMeta             the meta-data accessible to all consumers of the transaction.
+    *                                    See [[TransactionMeta]] for more information.
+    * @param transaction                 the submitted transaction. This transaction can contain local
+    *                                    contract-ids that need suffixing. The participant state may have to
+    *                                    suffix those contract-ids in order to guaranteed their global
+    *                                    uniqueness. See the Contract Id specification for more detail
+    *                                    daml-lf/spec/contract-id.rst.
+    * @param estimatedInterpretationCost Estimated cost of interpretation that may be used for
+    *                                    handling submitted transactions differently.
     * @return an async result of a SubmissionResult
     */
+  @silent("deprecated")
   def submitTransaction(
       submitterInfo: SubmitterInfo,
       transactionMeta: TransactionMeta,
       transaction: SubmittedTransaction,
-  ): CompletionStage[SubmissionResult]
+      estimatedInterpretationCost: Long,
```

Comment:
Because for a transaction you have interpretation cost whereas for non-transaction submissions you don't (and `CommitMetadata` is passed for each type).

Reply:
The relationship between submission type and metadata is not made explicit by an `Option` though; I'd consider specializing `CommitMetadata`.

Reply:
Agreed, that specialization will probably happen when we add more fields to `CommitMetadata`. Another angle is, however, that for Java consumers exposing different case classes may not be preferred and hence we may want to choose a simpler solution.